modules/dataframe/src/datastructures/dataframe.cpp

```diff
@@ -161,6 +161,24 @@ std::string DataFrame::getHeader(size_t idx) const { return columns_[idx]->getHe
 
 std::shared_ptr<const Column> DataFrame::getColumn(size_t index) const { return columns_[index]; }
 
+std::shared_ptr<Column> DataFrame::getColumn(const std::string &name) {
+    for (const auto col : columns_) {
+        if (col->getHeader() == name) {
+            return col;
+        }
+    }
+    return nullptr;
```

Comment:
Should we throw an exception here? I would prefer an exception here rather than a nullptr exception when I forgot to check if it exists ;)

Reply:
if you throw an exception you should return an reference not a pointer... you always have to check if the pointer is null. if it can't be null you should not return a pointer. 
should also use find_if preferably

Reply:
should be fixed now

Reply:
@petersteneteg Out of curiosity, why does it matter whether I use find or find_if? Both do exactly what I want.

Reply:
I guess the point here being that find_if already encapsulates this functionality (except for the comparison, hence find_if). And thus it's just a one liner.